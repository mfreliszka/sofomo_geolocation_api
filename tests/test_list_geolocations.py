"""Module containing tests for the get_geolocation endpoint."""

from fastapi import status
import pytest

from syrupy.assertion import SnapshotAssertion


@pytest.mark.anyio
async def test_list_existing_geolocations(
    app: pytest.fixture,
    IPGeolocation1_InDB_Model: pytest.fixture,
    IPGeolocation2_InDB_Model: pytest.fixture,
    snapshot: SnapshotAssertion,
    httpx_async_client: pytest.fixture,
):
    """Test retrieving a list of existing geolocations from the database."""
    response = await httpx_async_client.get(
        app.url_path_for("list_all_geolocations_from_database"),
    )

    assert response.json() == snapshot, "Actual response did not match snapshot"
    assert response.status_code == status.HTTP_200_OK, (
        f"Response status code: {response.status_code}"
    )
    assert response.headers["Content-Type"] == "application/json", (
        f"Response content type: {response.headers['Content-Type']}"
    )


@pytest.mark.anyio
async def test_list_existing_geolocations_with_limit(
    app: pytest.fixture,
    IPGeolocation1_InDB_Model: pytest.fixture,
    IPGeolocation2_InDB_Model: pytest.fixture,
    snapshot: SnapshotAssertion,
    httpx_async_client: pytest.fixture,
):
    """Test retrieving a list of existing geolocations from the database with limit."""
    response = await httpx_async_client.get(
        app.url_path_for("list_all_geolocations_from_database"),
        params={"limit": 1},
    )

    assert response.json() == snapshot, "Actual response did not match snapshot"
    assert response.status_code == status.HTTP_200_OK, (
        f"Response status code: {response.status_code}"
    )
    assert response.headers["Content-Type"] == "application/json", (
        f"Response content type: {response.headers['Content-Type']}"
    )


@pytest.mark.anyio
async def test_list_existing_geolocations_with_limit_and_offset(
    app: pytest.fixture,
    IPGeolocation1_InDB_Model: pytest.fixture,
    IPGeolocation2_InDB_Model: pytest.fixture,
    snapshot: SnapshotAssertion,
    httpx_async_client: pytest.fixture,
):
    """Test retrieving a list of existing geolocations from the database with limit and offset."""
    response = await httpx_async_client.get(
        app.url_path_for("list_all_geolocations_from_database"),
        params={"limit": 1, "offset": 1},
    )

    assert response.json() == snapshot, "Actual response did not match snapshot"
    assert response.status_code == status.HTTP_200_OK, (
        f"Response status code: {response.status_code}"
    )
    assert response.headers["Content-Type"] == "application/json", (
        f"Response content type: {response.headers['Content-Type']}"
    )


@pytest.mark.anyio
async def test_empty_list_of_geolocations(
    app: pytest.fixture,
    snapshot: SnapshotAssertion,
    httpx_async_client: pytest.fixture,
):
    """Test retrieving a list of existing geolocations from the database."""
    response = await httpx_async_client.get(
        app.url_path_for("list_all_geolocations_from_database"),
    )

    assert response.json() == snapshot, "Actual response did not match snapshot"
    assert response.status_code == status.HTTP_200_OK, (
        f"Response status code: {response.status_code}"
    )
    assert response.headers["Content-Type"] == "application/json", (
        f"Response content type: {response.headers['Content-Type']}"
    )
