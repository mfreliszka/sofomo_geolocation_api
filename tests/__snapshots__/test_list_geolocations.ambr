# serializer version: 1
# name: test_empty_list_of_geolocations[asyncio]
  dict({
    'items': list([
    ]),
    'limit': 10,
    'offset': 0,
    'total': 0,
  })
# ---
# name: test_empty_list_of_geolocations[trio]
  dict({
    'items': list([
    ]),
    'limit': 10,
    'offset': 0,
    'total': 0,
  })
# ---
# name: test_list_existing_geolocations[asyncio]
  dict({
    'items': list([
      dict({
        'city': 'Mountain View',
        'continent_code': 'NA',
        'continent_name': 'North America',
        'country_code': 'US',
        'country_name': 'United States',
        'id': 1,
        'ip': '8.8.8.8',
        'latitude': 37.422,
        'longitude': -122.084,
        'region_code': 'CA',
        'region_name': 'California',
        'type': 'IPv4',
        'updated_at': '2020-01-01T00:00:00',
        'zip': '94043',
      }),
      dict({
        'city': 'Mountain View',
        'continent_code': 'NA',
        'continent_name': 'North America',
        'country_code': 'US',
        'country_name': 'United States',
        'id': 2,
        'ip': '8.8.8.7',
        'latitude': 37.422,
        'longitude': -122.084,
        'region_code': 'OH',
        'region_name': 'Ohio',
        'type': 'IPv4',
        'updated_at': '2020-01-01T00:00:00',
        'zip': '94043',
      }),
    ]),
    'limit': 10,
    'offset': 0,
    'total': 2,
  })
# ---
# name: test_list_existing_geolocations[trio]
  dict({
    'items': list([
      dict({
        'city': 'Mountain View',
        'continent_code': 'NA',
        'continent_name': 'North America',
        'country_code': 'US',
        'country_name': 'United States',
        'id': 1,
        'ip': '8.8.8.8',
        'latitude': 37.422,
        'longitude': -122.084,
        'region_code': 'CA',
        'region_name': 'California',
        'type': 'IPv4',
        'updated_at': '2020-01-01T00:00:00',
        'zip': '94043',
      }),
      dict({
        'city': 'Mountain View',
        'continent_code': 'NA',
        'continent_name': 'North America',
        'country_code': 'US',
        'country_name': 'United States',
        'id': 2,
        'ip': '8.8.8.7',
        'latitude': 37.422,
        'longitude': -122.084,
        'region_code': 'OH',
        'region_name': 'Ohio',
        'type': 'IPv4',
        'updated_at': '2020-01-01T00:00:00',
        'zip': '94043',
      }),
    ]),
    'limit': 10,
    'offset': 0,
    'total': 2,
  })
# ---
# name: test_list_existing_geolocations_with_limit[asyncio]
  dict({
    'items': list([
      dict({
        'city': 'Mountain View',
        'continent_code': 'NA',
        'continent_name': 'North America',
        'country_code': 'US',
        'country_name': 'United States',
        'id': 1,
        'ip': '8.8.8.8',
        'latitude': 37.422,
        'longitude': -122.084,
        'region_code': 'CA',
        'region_name': 'California',
        'type': 'IPv4',
        'updated_at': '2020-01-01T00:00:00',
        'zip': '94043',
      }),
    ]),
    'limit': 1,
    'offset': 0,
    'total': 2,
  })
# ---
# name: test_list_existing_geolocations_with_limit[trio]
  dict({
    'items': list([
      dict({
        'city': 'Mountain View',
        'continent_code': 'NA',
        'continent_name': 'North America',
        'country_code': 'US',
        'country_name': 'United States',
        'id': 1,
        'ip': '8.8.8.8',
        'latitude': 37.422,
        'longitude': -122.084,
        'region_code': 'CA',
        'region_name': 'California',
        'type': 'IPv4',
        'updated_at': '2020-01-01T00:00:00',
        'zip': '94043',
      }),
    ]),
    'limit': 1,
    'offset': 0,
    'total': 2,
  })
# ---
# name: test_list_existing_geolocations_with_limit_and_offset[asyncio]
  dict({
    'items': list([
      dict({
        'city': 'Mountain View',
        'continent_code': 'NA',
        'continent_name': 'North America',
        'country_code': 'US',
        'country_name': 'United States',
        'id': 2,
        'ip': '8.8.8.7',
        'latitude': 37.422,
        'longitude': -122.084,
        'region_code': 'OH',
        'region_name': 'Ohio',
        'type': 'IPv4',
        'updated_at': '2020-01-01T00:00:00',
        'zip': '94043',
      }),
    ]),
    'limit': 1,
    'offset': 1,
    'total': 2,
  })
# ---
# name: test_list_existing_geolocations_with_limit_and_offset[trio]
  dict({
    'items': list([
      dict({
        'city': 'Mountain View',
        'continent_code': 'NA',
        'continent_name': 'North America',
        'country_code': 'US',
        'country_name': 'United States',
        'id': 2,
        'ip': '8.8.8.7',
        'latitude': 37.422,
        'longitude': -122.084,
        'region_code': 'OH',
        'region_name': 'Ohio',
        'type': 'IPv4',
        'updated_at': '2020-01-01T00:00:00',
        'zip': '94043',
      }),
    ]),
    'limit': 1,
    'offset': 1,
    'total': 2,
  })
# ---
